src/app.js

```diff
@@ -320,6 +347,8 @@ var run = function () {
 
     currentFile = file
 
+    files.event.trigger('fileFocus', [file])
```

Comment:
I think only a module itself should be able to trigger an event. I know this is already the case in app.js, but we need to change this. Please contain such uses to app.js, i.e. replace this by calling `switchToFile` inside app.js (provided as a callback from app.js).